lib/db/db.rb

```diff
@@ -9,26 +9,63 @@ def size
     all.size
   end
 
+  def all
+    reader.nil? ? [] : reader.run
+  end
+
+  def at(index)
+    empty || invalid(index) ? {} : all[index]
+  end
+
   def add(contact)
-    contacts = all
-    contacts << contact
+    contacts = all << format(contact)
```

Comment:
So, would it make sense to extract it to a separate method?
Assigning it to a variable and having a mutation (`<<`) happening makes it harder to understand what's happening. As a reader of the code I don't want to look up the operator precedence rules to know what's actually being assigned.
It boils down to _don't make your readers think_ ;)